### PR TITLE
docs: adding reference guide

### LINKED ISSUE #142: A demo application with env-schema for reference - should we mention anywhere here to help others?
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Hey folks,

`env-schema` is great. I'm building a repository that demonstrates a fully functional Fastify application with `env-schema`, `fastify/env`, and `dotenv`. Would it be relevant to link it here anywhere in this or other repositories?

### PR DESCRIPTION
**Pull request by lirantal**
Adding a section to list helpful resource guides to get started with building with env-schema

Fixes #142 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Could you split creation of the Fastify application with the call to listen in two files, as this enables testing without using globals?

**Comment by lirantal** [comment]
Sure thing. I was going for simplicity there but have merged a PR to change that to a more modular structure: https://github.com/lirantal/fastify-dotenv-envschema-example/pull/1 (landed in `main` already)

File: README.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 Utility to check environment variables using [JSON schema](https://json-schema.org/), [Ajv](http://npm.im/ajv), and
 [dotenv](http://npm.im/dotenv).
 
+See [supporting resources](#supporting-resources) section for helpful guides on getting started
```
Comment (Fdawgs) [inline README.md:10]:
```suggestion
See [supporting resources](#supporting-resources) section for helpful guides on getting started.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (lirantal) [reply README.md]: Good catch!

**Review by Fdawgs — APPROVED** [review]

**Review by climba03003 — COMMENTED** [review]
Shouldn't it place in `@fastify/env` when you did not directly using `env-schema`?

**Comment by lirantal** [comment]
It's a good question and I was asking myself the same thing but I thought the ideal place is here because folks using the `@fastify/env` plugin would probably already know best practices and how to use it, yet new folks landing on this repository might look for how to get started and how they can use `env-schema`, especially since it doesn't have anything to do with Fastify as the app framework itself, but just happens to live on this org.

So, I'd keep it here to be more helpful but happy to move elsewhere if you think it'd be a better fit.

**Comment by mcollina** [comment]
I personally find env-schema better and I don't use fastify-env ;).

**Comment by lirantal** [comment]
fastify-env is pretty small and is anyway just a wrapper around env-schema though, isn't it? I figured if you don't use fastify-env you probably just do the same plugging wrapping yourself.

**Comment by mcollina** [comment]
I tend to _not_ wrap this into a plugin, but rather read it before Fastify is started. In this way I can customize the options of Fastify itself

**Comment by jsumners** [comment]
> I tend to _not_ wrap this into a plugin, but rather read it before Fastify is started. In this way I can customize the options of Fastify itself

Agreed. I really don't understand the point of `fastify-env`. 

```js
'use strict'

const dotenv = require('dotenv')
const config = dotenv.config()

const fastify = require('fastify')({ /* use stuff from config */})
fastify.decorate('env', {
  getter() { return config }
})
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**